cloudwatch_event_rules.tf

```diff
@@ -29,4 +29,8 @@ resource "aws_cloudwatch_event_target" "lambda" {
   arn            = lookup(each.value, "cloudwatch_event_target_arn", local.function_arn)
   rule           = aws_cloudwatch_event_rule.lambda[each.key].name
   input          = lookup(each.value, "cloudwatch_event_target_input", null)
+  retry_policy {
```

Comment:
Nice 👍 

Could you please add a new line before `retry_policy` and enhance the [scheduled example](https://github.com/moritzzimmer/terraform-aws-lambda/blob/30e15b6caf45fd05be89d0671b7883d369f82d93/examples/with-cloudwatch-event-rules/main.tf#L21) with the new config options.

It seems like a once configured `maximum_event_age_in_seconds` can't be removed by removing `cloudwatch_event_target_retry_policy_maximum_event_age_in_seconds`. The next apply fails with an exception like this:

```
Error: updating EventBridge Target (terraform-20240315143915351100000003-terraform-20240315143915805900000006): InvalidParameter: 1 validation error(s) found.
│ - minimum field value of 60, PutTargetsInput.Targets[0].RetryPolicy.MaximumEventAgeInSeconds.
```

As a workaround users would need to specify a valid value or remove the whole event rule. This seems like an issue in the aws provider - what do you think?

Reply:
I updated it so that, if unset, it'll use the documented default. Tell me what you think about that approach. Also updated example.

Reply:
@elliottohara I also thought about this as a workaround. As a consquence users will now always define a retry policy. According to the [aws docs](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-rule-dlq.html) the defaults are 24h for the `maximum_event_age_in_seconds` and 185 for `maximum_retry_attempts`. I'd say we use those as defaults inline (no need to declare the as locals).

aws provider output while testing to confirm max values:

```
// range value maximum_event_age_in_seconds
Error: expected retry_policy.0.maximum_event_age_in_seconds to be in the range (0 - 86400), got 90000

// min value maximum_event_age_in_seconds
- minimum field value of 60, PutTargetsInput.Targets[0].RetryPolicy.MaximumEventAgeInSeconds.

// range maximum_retry_attempts
Error: expected retry_policy.0.maximum_retry_attempts to be in the range (0 - 185), got 186
```